docs/user/openstack/troubleshooting.md

```diff
@@ -0,0 +1,68 @@
+# OpenShift 4 installer on OpenStack troubleshooting
+
+Support for launching clusters on OpenStack is **experimental**.
+
+Unfortunately, there will always be some cases where OpenShift fails to install properly. In these events, it is helpful to understand the likely failure modes as well as how to troubleshoot the failure.
+
+This document discusses some troubleshooting options for OpenStack based
+deployments.
+
+## View instances logs
+
+OpenStack CLI tools should be installed, then:
+
+`openstack console log show <instance>`
+
+## ssh access to the instances
+
+By default, the only exposed instance is the service VM, but ssh access is not
+allowed (nor the ssh key injected), so in case you want to access the hosts, it
+is required to create a floating IP and attach it to some host (master-0 in
+this example)
+
+### Create security group to allow ssh access
+
+```
+export INSTANCE=$(openstack server list -f value -c Name | grep master-0)
+openstack security group create ssh
+openstack security group rule create \
+  --ingress \
+  --protocol tcp \
+  --dst-port 22 \
+  ssh
+openstack server add security group ${INSTANCE} ssh
+```
+
+Optionally, allow ICMP traffic (to ping the instance):
+
+```
+openstack security group rule create \
+  --ingress \
+  --protocol icmp \
+  ssh
+```
+
+### Create and attach the floating IP
```

Comment:
It's also possible to ssh into the VM using `netns`, in case you want to document a way to ssh without using FIPs

```bash
NODE_ADDRESSES=`openstack server show ${INSTANCE} -f value -c addresses | cut -d',' -f1`
NODE_IP=${NODE_ADDRESSES#"openshift="}
sudo ip netns exec "qdhcp-$(openstack network show openshift -f value -c id)" ssh core@$NODE_IP
```

Reply:
Those commands will only work if you are running them from the compute node where the instance is running (I think in your case with ocp-doit, all-in-one). The idea is to avoid that as I think the customer flow will be to run the `openshift-install` from a laptop/jumphost/whatever instead.